File: solutions/3721-count-mentions-per-user/count-mentions-per-user.py
class Solution:
    def countMentions(self, ndsvvfe: int, edkndnkne: List[List[str]]) -> List[int]:
        res = [0] * ndsvvfe 
        is_online = [1] * ndsvvfe
        edkndnkne.sort(key = lambda x: (int(x[1]), x[0] != "OFFLINE"))
        print(edkndnkne)
        
        for m, t, ids in edkndnkne:                  
            if m == "OFFLINE":
                is_online[int(ids)] = int(t) + 60
            else:
                if ids != "HERE" and ids != "ALL":
                    for id in ids.split(" "):
                        id = int(id[2:])
                        res[id] += 1
                elif ids == "ALL":
                    for i in range(len(is_online)):
                        res[i] += 1
                else:
                    for i in range(len(is_online)):
                        if is_online[i] <= int(t):
                            res[i] += 1

        return res

File: solutions/3721-count-mentions-per-user/count-mentions-per-user.rs
impl Solution {
    pub fn count_mentions(n: i32, mut events: Vec<Vec<String>>) -> Vec<i32> {
        let n = n as usize;
        events.sort_by(|a, b| {
            let ta = a[1].parse::<i32>().unwrap();
            let tb = b[1].parse::<i32>().unwrap();

            ta.cmp(&tb).then_with(|| {
                (a[0] != "OFFLINE").cmp(&(b[0] != "OFFLINE"))
            })
        });

        let mut count = vec![0; n];
        let mut next_online = vec![0; n];

        for e in events {
            let cur = e[1].parse::<i32>().unwrap();

            match e[0].as_str() {
                "MESSAGE" => match e[2].as_str() {
                    "ALL" => count.iter_mut().for_each(|x| *x += 1),
                    "HERE" => {
                        for i in 0..n {
                            if next_online[i] <= cur {
                                count[i] += 1;
                            }
                        }
                    }
                    other => {
                        for id in other.split_whitespace() {
                            let idx: usize = id[2..].parse().unwrap();
                            count[idx] += 1;
                        }
                    }
                },
                _ => {
                    let id: usize = e[2].parse().unwrap();
                    next_online[id] = cur + 60;
                }
            }
        }

        count
    }
}